controller/update.go

```diff
@@ -300,11 +300,16 @@ func (c controller) UpdateWithStrategy(ctx context.Context, req Request, opts Up
 		newReq.SliceIDs = []string{sliceID}
 
 		for {
-			currentSliceReq := req
-
 			c.Config.Logger.Debug(ctx, "controller: attempting to add slice: %v", sliceID)
-			// add
-			maxGrowth := opts.MaxGrowth + numTotal - opts.MinAlive - int(addInProgress)
+
+			currentSliceReq := req
+			// Calculating the number of groups allowed to be added during this
+			// iteration respects the total number of running groups. This is because
+			// we later going to check how many are actually running after the
+			// addition. So the next iteration will calculate and check again and
+			// respect the total number of running groups. See also
+			// isGroupAdditionAllowed.
```

Comment:
The calculation for `minAlive` is more straight forward. I would keep it as it is. There was no problem. I would not know what to document there besides the rather obvious information.